api-pre2.2/src/main/java/org/openmrs/module/coreapps/htmlformentry/EncounterDiagnosesElement.java

```diff
@@ -136,6 +142,39 @@ public String generateHtml(FormEntryContext context) {
             }
         }
     }
+    
+    /**
+     * This method recieves a comma seperated list of diagnosis sets which are verified and then
+     * returns as comma seperated list of their uuids. diagnosisSetIds may be concept's 1). uuid
+     * 2). mapping, 3). id
+     *
+     * @param diagnosisSetIds
+     * @return uuids
+     */
+    private String validateAndFormat(String diagnosisSetIds) {
```

Comment:
Why is this here again?

Reply:
Yes, of the three `EncounterDiagnoses*` on is for the `EncounterDiagnosesByObsElement`, and then for the other two `EncounterDiagnosesElement`s, one of them is used by versions of core less than 2.2 .